tests/unit/acquisition/test_function.py

```diff
@@ -461,6 +471,175 @@ def prepare_acquisition_function(
     npt.assert_allclose(eci(x), ei(x) * pof(x))
 
 
+class _PseudoTrainableQuadratic(QuadraticMeanAndRBFKernel, PseudoTrainableProbModel):
+    pass
+
+
+class _PseudoTrainableLinear(GaussianProcess, PseudoTrainableProbModel):
+    def __init__(
+        self,
+        *,
+        kernel_amplitude: float | TensorType | None = None,
+    ):
+        kernel = tfp.math.psd_kernels.ExponentiatedQuadratic(kernel_amplitude)
+        super().__init__([lambda x: tf.reduce_sum(x, axis=-1, keepdims=True)], [kernel])
+
+    def __repr__(self) -> str:
+        return "_PseudoTrainableLinear()"
+
+
+_mo_test_models = (
+    _PseudoTrainableQuadratic,
+    _PseudoTrainableLinear,
+    _PseudoTrainableQuadratic,
+)
+
+
+def test_ehvi_builder_raises_for_empty_data() -> None:
+    num_obj = 3
+    dataset = Dataset(tf.zeros([0, 2]), tf.zeros([0, 1]))
+    model = ModelStack(*[(_PseudoTrainableQuadratic(), 1) for _ in range(num_obj)])
+
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        ExpectedHypervolumeImprovement().prepare_acquisition_function(dataset, model)
+
+
+def test_ehvi_builder_builds_expected_hv_improvement_using_pareto_from_model() -> None:
+    num_obj = 2
+    train_x = tf.constant([[-2.0], [-1.5], [-1.0], [0.0], [0.5], [1.0], [1.5], [2.0]])
+    dataset = Dataset(
+        train_x,
+        tf.tile(
+            tf.constant([[4.1], [0.9], [1.2], [0.1], [-8.8], [1.1], [2.1], [3.9]]), [1, num_obj]
+        ),
+    )
+
+    model = ModelStack(*[(_mo_test_models[_](), 1) for _ in range(num_obj)])
+    acq_fn = ExpectedHypervolumeImprovement().prepare_acquisition_function(dataset, model)
+
+    model_pred_observation = model.predict(train_x)[0]
+    _prt = Pareto(model_pred_observation)
+    xs = tf.linspace([[-10.0]], [[10.0]], 100)
+    expected = expected_hv_improvement(model, _prt, get_reference_point(_prt.front))(xs)
+    npt.assert_allclose(acq_fn(xs), expected)
+
+
+@pytest.mark.parametrize("at", [tf.constant([[0.0], [1.0]]), tf.constant([[[0.0], [1.0]]])])
+def test_ehvi_raises_for_invalid_batch_size(at: TensorType) -> None:
+    num_obj = 2
+    train_x = tf.constant([[-2.0], [-1.5], [-1.0], [0.0], [0.5], [1.0], [1.5], [2.0]])
+
+    model = ModelStack(*[(_mo_test_models[_](), 1) for _ in range(num_obj)])
```

Comment:
general comment: underscore in python (mostly) refers to a throwaway variable, so it reads odd to then use it

Reply:
Changed.